servant-server/src/Servant/Server/Internal/RoutingApplication.hs

```diff
@@ -264,18 +303,17 @@ passToServer Delayed{..} x =
 runDelayed :: Delayed env a
            -> env
            -> Request
-           -> CleanupRef
-           -> IO (RouteResult a)
-runDelayed Delayed{..} env req cleanupRef =
+           -> ResourceT IO (RouteResult a)
+runDelayed Delayed{..} env req  =
   runDelayedIO
     (do c <- capturesD env
         methodD
         a <- authD
         b <- bodyD
-        DelayedIO $ \ r _cleanup -> return (serverD c a b r)
+        r <- ask
+        returnRouteResult (serverD c a b r)
```

Comment:
`Handler` cannot, it's just `ExceptT ServerErr IO a`. Then we'll need to add `ResourceT` there too. I'm neutral about that.